components/dashboard/src/teams/TeamUsageBasedBilling.tsx

```diff
@@ -326,6 +334,9 @@ function CreditCardInputForm() {
         }
         setIsLoading(true);
         try {
+            // Create Stripe customer for team & currency (or update currency)
+            await getGitpodService().server.createOrUpdateStripeCustomerForTeam(props.teamId, currency);
+            // FIXME(janx): The above returns too early, when the customer doesn't exist yet? 🤔
```

Comment:
This breaks the first sign-up attempt:
1. Add credit card
2. Fails after reload 💥 (server error "Error: No Stripe customer profile for team '9d19a7ed-0154-43f8-a4c5-2dea787d06d8'")
3. Add credit card again
4. Now it works

Reply:
Looks like a race condition:

- This call is completed by `08:11:40.128Z`:

```ts
this.getStripe().customers.create({
    email: User.getPrimaryEmail(user),
    name: userName ? `${userName} (${team.name})` : team.name,
    metadata: {
        teamId: team.id,
    },
});
```

- This call at `08:11:43.554Z` comes back empty (no results):

```ts
this.getStripe().customers.search({
    query: `metadata['teamId']:'${teamId}'`,
});
```

- The same call at `08:15:27.202Z` comes back with one result (the created customer)

Maybe we need to poll Stripe after creating a customer?

Reply:
Polling fixed it. ✅